.github/workflows/pnp-service-definition-validate.yml

```diff
@@ -0,0 +1,39 @@
+name: Validate Service Definition File(s)
+on:
+  workflow_call:
+    inputs:
+      service-type:
+        description: 'Type of the service.'
+        type: string
+        required: true
+      service-definitions:
+        description: 'List of service definition file paths provided as a JSON array string. For example: ["./folder1/service-definition1.yaml", "./folder2/service-definition2"].'
+        type: string
+        required: true
+
+jobs:
+  validate-service-definition:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Configure Git
+        uses: extenda/actions/setup-git@v0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Fetch latest changes
+        run: git pull origin ${{ github.ref }}
```

Comment:
If you set `fetch-depth: 0` for `checkout` action, it will fetch all history for all branches. There are other parameters as well that might be use to configure the check-out process. Have you looked at it? Maybe this step is not needed at all?

Reply:
Yeah, this step is redundant. Just using `checkout` action without specifying additional parameters is enough here.